spec/tapioca/compilers/dsl/protobuf_spec.rb

```diff
@@ -0,0 +1,216 @@
+            end
+          end
+        end
+
+        Cart = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart").msgclass
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Cart
+          sig { returns(Google::Protobuf::UInt64Value) }
+          def cart_item_index; end
+
+          sig { params(value: Google::Protobuf::UInt64Value).returns(Google::Protobuf::UInt64Value) }
+          def cart_item_index=(value); end
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content))
+    end
+
+    it("generates methods in RBI files for classes with Protobuf with enum field") do
+      content = <<~RUBY
+        require 'google/protobuf/timestamp_pb'
+        require 'google/protobuf/wrappers_pb'
+        Google::Protobuf::DescriptorPool.generated_pool.build do
+          add_file("cart.proto", :syntax => :proto3) do
+            add_message "MyCart" do
+              optional :value_type, :enum, 1, "VALUE_TYPE"
+            end
+            add_enum "VALUE_TYPE" do
+              value :NULL, 0
+              value :FIXED_AMOUNT, 1
+              value :PERCENTAGE, 2
+            end
+          end
+        end
+        Cart = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart").msgclass
+        Cart::VALUE_TYPE = Google::Protobuf::DescriptorPool.generated_pool.lookup("VALUE_TYPE").enummodule
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Cart
+          sig { returns(Cart::VALUE_TYPE) }
+          def value_type; end
+
+          sig { params(value: Cart::VALUE_TYPE).returns(Cart::VALUE_TYPE) }
+          def value_type=(value); end
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content))
+    end
+
+    it("generates methods in RBI files for classes with Protobuf with enum field with defined type") do
+      content = <<~RUBY
+        require 'google/protobuf/timestamp_pb'
+        require 'google/protobuf/wrappers_pb'
+        Google::Protobuf::DescriptorPool.generated_pool.build do
+          add_file("cart.proto", :syntax => :proto3) do
+            add_message "MyCart" do
+              optional :value_type, :enum, 1, "MyCart.MYVALUETYPE"
+            end
+            add_enum "MyCart.MYVALUETYPE" do
+              value :ACROSS, 0
+              value :ONE, 1
+              value :EACH, 2
+            end
+          end
+        end
+        Cart = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart").msgclass
+        Cart::MYVALUETYPE = Google::Protobuf::DescriptorPool.generated_pool.lookup("MyCart.MYVALUETYPE").enummodule
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Cart
+          sig { returns(Cart::MYVALUETYPE) }
+          def value_type; end
+
+          sig { params(value: Cart::MYVALUETYPE).returns(Cart::MYVALUETYPE) }
+          def value_type=(value); end
+        end
+      RUBY
+
+      assert_equal(expected, rbi_for(content))
+    end
```

Comment:
Can we add a test that exercises all the different protobuf types that we map to Ruby types? One `Cart` message that has `int32_field`, `int64_field`, `uint32_field`, etc etc, enumerating all the types that we map would be enough. And you can just match the generated setter methods to assert that we figure out the correct Ruby types for them.